### PR TITLE
Implement Depth Excess acoustic variable

### PR DESCRIPTION
**Pull request by htmlboss**
## Background

* Allow ingestion of optional bathymetry files.
* Calculation layer has new Depth Excess function to display difference between deep sound channel bottom and bathymetry bottom. Only model-specific bathymetry files are supported (i.e. there is no interpolation method).

## Why did you take this approach?
Seems like the quickest and simplest implementation method.

## Anything in particular that should be highlighted?

* Am I doing this right?
* Colour scale values.

## Screenshot(s)

image

## Checks
- [ ] I ran unit tests.
- [x] I've tested the relevant changes from a user POV.

_Hint_ To run all python unit tests run the following command from the root repository directory:
```sh
bash run_python_tests.sh
```

### CODE REVIEW

**Comment by htmlboss** [comment]
Ignore the branch name lol.

**Comment by htmlboss** [comment]
screenshot pending

File: data/calculated_parser/functions.py [hunk 1]
```diff
@@ -386,6 +386,27 @@ def deepsoundchannelbottom(depth, latitude, temperature, salinity) -> np.ndarray
...
+
+        * depth: Depth in meters
+
+        * latitude: Latitude in degrees North
+
+        * temperature: Temperatures in Celsius
+
+        * salinity: Salinity
+
+        * bathy:
```
Comment (htmlboss) [inline data/calculated_parser/functions.py:403]: todo: document this

**Review by nsoontie — COMMENTED** [review]
Looks good Nabil. Nice and simple.

I suppose there is a datasetconfig.json update to that corresponds to these changes?

I formatted the bathy files for the RIOPS and GIOPS user grids. You can find them here:
RIOPS polar stereographic - hank:/tank/data/grids/ecc/riops/bathy_meter_riops_ps5km_reformatted-ONav.nc

GIOPS latlon - hank:/tank/data/grids/ecc/giops/bathy_meter_giops_latlon_reformatted-ONav.nc
GIOPS polar stereographic - hank:/tank/data/grids/ecc/giops/bathy_meter_giops_ps5km_reformatted-ONav.nc

The nco commands used to create these files are in the history attribute.

**Comment by dwayne-hart** [comment]
@htmlboss please update your branch as it is currently out of date.

**Comment by nsoontie** [comment]
@dwayne-hart Did we intend to leave this pull request out of the 6.5.7-rc0 release candidate?

On thes taging site, I see the Depth Excess variable is available for dataset 01 but it doesn't display anything likely because this pull request hasn't been merged.

**Review by dwayne-hart — APPROVED** [review]

**Comment by dwayne-hart** [comment]
@nsoontie, the branch was not up to date. I did request @htmlboss to correct the issue. I have performed that task and have merged in the code.

**Pull request merged**